### ISSUE TITLE
Built in operator support

### ISSUE THREAD

**Issue by orhanbalci** [m-decoster/RsGenetic#30]
Hi there. 
It would be nice to have pluggable crossover and mutation operators and implementations of well known  operators.

**Comment by m-decoster** [comment]
Could you give an example of what you would like to see?

**Comment by orhanbalci** [comment]
User should be able to use same phenotype implementation with different crossover or mutation operators. With the current state of the library one should write separate phenotype implementations for each operator.

**Comment by m-decoster** [comment]
I understand where you're coming from. Implementing this feature could potentially impact the API of the library a lot, and in my opinion (feel free to disagree or persuade me otherwise) is better suited for a version 2.0 of the library (see also #23).

In the meantime, I just pushed an example (7e2b959b5c22cf6e3c6f41d2d66eed00a63d41f4) of how you can work around this in a fairly ergonomic manner.

I would gladly implement your suggestion, but I don't think I'll find the time to do it very soon.

**Comment by orhanbalci** [comment]
You are right. This would require API breaking changes. One implementation idea is to use associated types for these operators.
